changelog/destroy_noinit.dd

```diff
@@ -0,0 +1,3 @@
+Added `initialize` template argument to `object.destroy()`.
+
+`object.destroy()` now receives an `initialize` argument to specify whether to re-initialize the object after destruction.
```

Comment:
Then perhaps call it `reinitialize` (or maybe `reinit_memory`)?

Reply:
I considered `reinitialize`, I decided against it. The other one feels unnecessarily wordy to me.
3rd opinions anyone?